ATMoviess/ATMoviess/Services/UpcomingMoviesService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using ATMoviess.Models;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ATMoviess.Services
+{
+    public class MoviesService
+    {
+        private List<Genre> GenresList { get; set; }
+        
+        /// <summary>
+        /// Gets the list of genres
+        /// </summary>
+        /// <returns></returns>
+        private async Task GetGenresAsync()
+        {
+            var parameters = new Dictionary<string, object>();
+            parameters.Add("language", "en-US");
+
+            var content = await CommunicationService.GetAsync("genre/movie/list", parameters);
+            var response = await content.Content.ReadAsStringAsync();
+
+            GenresList = JsonConvert.DeserializeObject<GenresListModel>(response).Genres;
+        }
+        
+        /// <summary>
+        /// Gets the the whole list of upcoming movies from the TMDB database.
+        /// Orders the list by release date. Does not filter region.
+        /// </summary>
+        /// <returns>Ordered list of upcoming movies</returns>
+        public async Task<List<Result>> GetUpcomingMoviesAsync()
+        {
+            var parameters = new Dictionary<string, object>();
+            parameters.Add("language", "en-US");
+            parameters.Add("page", 1);
+
+            var content = await CommunicationService.GetAsync("movie/upcoming", parameters);
+            var response = await content.Content.ReadAsStringAsync();
+
+            var result = JsonConvert.DeserializeObject<UpcomingMoviesModel>(response);
+
+            for (int i = 2; i <= result.Total_pages; i++)
```

Comment:
I see. The problem was that the upcoming movies endpoint was returning movies with release dates before "today" date, because of how the API works with region releases.
The only way I thought of doing that was fetching the whole list and them filtering it by today's date and up.
I could have filtered by region, but I didn't think it was good enough to only show that to the user.
The infinite list idea is great, but it wouldn't order the movies by release date, and also wouldn't filter them by release dates like I said before.